Sources/SwiftProtobuf/ProtobufBinaryTypes.swift

```diff
@@ -789,14 +777,21 @@ public extension ProtobufBinaryMessageBase {
 
 public extension ProtobufMessage {
     static func decodeProtobufMapValue(decoder: inout ProtobufFieldDecoder, value: inout Self?) throws {
-        let handled = try decoder.decodeSingularMessageField(fieldType: Self.self, value: &value)
-        assert(handled)
+        try decoder.decodeSingularMessageField(fieldType: Self.self, value: &value)
+        assert(value != nil)
     }
 
     init(protobuf: Data) throws {
         try self.init(protobuf: protobuf, extensions: nil)
     }
 
+    init(protobufBuffer: UnsafeBufferPointer<UInt8>) throws {
+        try self.init(protobufBuffer: protobufBuffer, extensions: nil)
+    }
+}
+
+// For Proto3 messages, we need to preserve the unknown fields when we're done
```

Comment:
The alternative is basically to generate `decodeIntoSelf`.  I considered that, but...

1. I'm trying to reduce boilerplate in the generated code.

2. There is a certain sense to having message structs explicitly tagged as being Proto2 or Proto3, since there are real semantic differences between the two.